### PR TITLE
(1164/1294) Update UI to match data model changes

### PR DESCRIPTION
**Pull request by yndajas**
## Context

https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/254 and https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/255 were the first steps in some changes to course-related entities on the API: moving the `referable` property to the `Offering` entity and moving to only have one `audience` per `Course`

https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/258, https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/262 and https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/267 update the seed data

## Changes in this PR

(Apologies for the large commits here - everything kind of needs updating at once due to the type changes!)

- Move `referable` property to `CourseOffering`
- Make a `Course`'s `audiences` singular: `audience`
- Ensure there are still enough referrals of a particular combination of fields to fully smoke test pagination
- Two minor improvements/tidy ups

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

**Comment by yndajas** [comment]
[Read `referable` from an `Offering` rather than a `Course` (S)](https://trello.com/c/hXhBpovQ/1164-read-referable-from-an-offering-rather-than-a-course-s)

**Comment by yndajas** [comment]
[Read singular "audience" value from API (M)](https://trello.com/c/RlhskYHn/1294-read-singular-audience-value-from-api-m)

File: integration_tests/e2e/find.cy.ts [hunk 1]
```diff
@@ -131,36 +133,44 @@ context('Find', () => {
...
         courseOfferingPage.shouldContainMakeAReferralButtonLink()
       })
     })
 
-    describe('and a non-referable course', () => {
+    describe('and a non-referable course offering', () => {
       it('does not show the "Make a referral" button on an offering', () => {
-        const nonReferableCourse = courseFactory.build({ referable: false })
-        cy.task('stubCourseByOffering', { course: nonReferableCourse, courseOfferingId: courseOffering.id })
+        const referableCourseOffering = courseOfferingFactory.build({
```
Comment (jsrobertson) [inline integration_tests/e2e/find.cy.ts:162]: `nonReferableCourseOffering`?

File: script/utils/generateApiSeeds/tableRecords.ts [hunk 2]
```diff
@@ -89,22 +84,22 @@ export default class TableRecords {
...
-            firstOfferingForActiveCaseLoadId.id,
+            firstReferableOfferingForActiveCaseLoadId.id,
             faker.helpers.arrayElement(prisoners).prisonerNumber,
           ),
         )
       }
 
       // some other offerings for active case load ID should have referrals
-      const nonFirstOfferingsForActiveCaseLoadId = this.offeringsForActiveCaseLoadId().slice(1)
+      const nonFirstReferableOfferingsForActiveCaseLoadId = this.referableOfferingRecordsForActiveCaseLoadId().slice(1)
```
Comment (jsrobertson) [inline script/utils/generateApiSeeds/tableRecords.ts:87]: Not a huge problem but wonder if there's a better word/phrase than`nonFirst`? `subsequent`?

File: server/views/partials/audienceTag.njk [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{% from "govuk/components/tag/macro.njk" import govukTag %}
+
+{% macro audienceTag(tag) %}
+  <p class="govuk-body">
```
Comment (jsrobertson) [inline server/views/partials/audienceTag.njk:4]: I forget that we don't need the `govuk-body` class as we have `$govuk-global-styles: true;` set.  I know this is still used in a few other places and won't break anything so happy to leave and they can be removed as and when we make changes in the future.

**Review by jsrobertson — APPROVED** [review]
🙌 Nice.  Could of minor comments but nothing blocking.

File: script/utils/generateApiSeeds/tableRecords.ts (hunk 2 above)
Comment (yndajas) [reply script/utils/generateApiSeeds/tableRecords.ts]: That's way better!

File: server/views/partials/audienceTag.njk (hunk 3 above)
Comment (yndajas) [reply server/views/partials/audienceTag.njk:4]: Oh nice - didn't know about that. It looks like `govuk-link` might also be unnecessary: https://github.com/alphagov/govuk-frontend/blob/8eeca8e8a7980dfaceb6619ca97084fdb63634f6/packages/govuk-frontend/src/govuk/core/_global-styles.scss#L4-L12

I'll leave it for now though and add a card for it, as some of the integration test selectors might need updating

**Pull request merged**